File: Lottery.Api.WebHost/Services/LotteryService.cs
using Lottery.Api.Contracts;
using System;
using System.Linq;
using System.Collections.Generic;

namespace Lottery.Api.Application
{
    public class LotteryService : ILotteryService
    { 
        private ILotteryRepository _repository;

        // Could derive these values using a settings provider from .config or .json 
        private int _primaryNumberCount = 49;
        private int _totalAmountOfPrimaryNumbers = 5;
        private int _secondaryNumberCount = 10;
        private int _totalAmountOfSecondaryNumbers = 2;

        public LotteryService(ILotteryRepository repository)
        {
            _repository = repository;
        }

        public void CreateDraw(CreateLotteryDrawCommand lotteryDrawCommand)
        {
            var existingDraw = _repository.FindByName(lotteryDrawCommand.Name);
            if (existingDraw != null)
            {
                throw new ArgumentException($"Lottery draw {lotteryDrawCommand.Name} already exists");
            }

            Domain.LotteryDraw draw = new Domain.LotteryDraw(lotteryDrawCommand.Name,
                                            lotteryDrawCommand.Description,
                                            lotteryDrawCommand.DrawDate.Date,
                                            new Domain.LotteryNumbers(_primaryNumberCount, _totalAmountOfPrimaryNumbers),
                                            new Domain.LotteryNumbers(_secondaryNumberCount, _totalAmountOfSecondaryNumbers));
            _repository.Create(draw);
        }
        
        public void SpecifyWinningNumbers(string name, SpecifyWinningNumbersCommand winningNumbersCommand)
        {
            var existingDraw = _repository.FindByName(name);

            if (existingDraw == null)
            {
                throw new ArgumentException($"Unable to find lottery draw {name}");
            }

            existingDraw.SpecifyWinningNumbers(new Domain.WinningNumbers
            {
                Primary = winningNumbersCommand.Primary,
                Secondary = winningNumbersCommand.Secondary
            });
            _repository.Update(existingDraw);
        }

        public LotteryResultByDrawDateResult FindByDrawDate(DateTime drawDate)
        {
            var result = _repository.FindByDate(drawDate).FirstOrDefault();

            if (result != null) {

                return new LotteryResultByDrawDateResult {
                    Name = result.Name,
                    Description = result.Description,
                    DrawDate = result.DrawDate,
                    PrimaryNumbers = result.PrimaryNumbers.Numbers,
                    TotalAmountOfPrimaryNumbers = result.PrimaryNumbers.TotalAmountOfNumbers,
                    SecondaryNumbers = result.SecondaryNumbers.Numbers,
                    TotalAmountOfSecondaryNumbers = result.SecondaryNumbers.TotalAmountOfNumbers,
                    WinningPrimaryNumbers = result.WinningNumbers?.Primary ?? new List<int>(),
                    WinningSecondaryNumbers = result.WinningNumbers?.Secondary ?? new List<int>()
                };
            }

            return null;
        }
    }
}

File: Lottery.Api.WebHost/wwwroot/scripts/lottery.service.js
(function () {
    'use strict';

    angular
        .module('lottery')
        .factory('lotteryService', lotteryService);

    lotteryService.$inject = ['$http'];

    function lotteryService($http) {
        var svc = this,
            service = {
                createDraw: createDraw,
                findByDrawDate: findByDrawDate,
                specifyWinningNumbers: specifyWinningNumbers,
                handleError: handleError
            };

        svc.http = $http;

        function handleError(response) {
            alert(response.data);
        }

        function createDraw(lotteryDraw) {
            return svc.http.post('api/lottery/draw', lotteryDraw);
        };

        function getConfig() {
            var config = {
                primaryNumbers: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49],
                secondaryNumbers: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
            };

            return {
                primary: {
                    maxWinningNumbers: 6,
                    numbers: config.primaryNumbers.map(function (n) { return { value: n, isSelected: false }; })
                },
                secondary: {
                    maxWinningNumbers: 2,
                    numbers: config.secondaryNumbers.map(function (n) { return { value: n, isSelected: false }; })
                }
            };
        }

        function findByDrawDate(drawDate) {
            var d = new Date(drawDate)

            return svc.http.get('api/lottery/draw/' + d.toJSON());
        };

        function specifyWinningNumbers(model) {
            return svc.http.put('api/lottery/draw/' + escape(model.name) + '/winningNumbers', { primary: model.winningPrimaryNumbers, secondary: model.winningSecondaryNumbers });
        };

        return service;
    }

})();